cmd/backup-manager/app/clean/clean.go

```diff
@@ -43,22 +51,45 @@ func (bo *Options) cleanBRRemoteBackupData(ctx context.Context, backup *v1alpha1
 		return err
 	}
 	defer s.Close()
-	iter := s.List(nil)
+
+	iter := util.ListPage(s, nil)
 	for {
-		obj, err := iter.Next(ctx)
+		// list one page of object
+		objs, err := iter.Next(ctx, pageSize)
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
 			return err
 		}
-		klog.Infof("Prepare to delete %s for cluster %s", obj.Key, bo)
-		err = s.Delete(context.Background(), obj.Key)
-		if err != nil {
-			return err
+
+		// batch delete objects
+		var s3cli *s3.S3
+		var result *util.BatchDeleteObjectsResult
+		if ok := s.As(&s3cli); ok {
+			klog.Infof("Delete a page of object by S3 batch delete")
+			bucket := backup.Spec.StorageProvider.S3.Bucket
+			prefix := backup.Spec.StorageProvider.S3.Prefix
+			for i := range objs {
+				objs[i].Key = fmt.Sprintf("%s/%s", prefix, objs[i].Key)
+			}
+			result = util.BatchDeleteObjectsOfS3(ctx, s3cli, bucket, objs, pageConcurrency)
+		} else {
+			klog.Infof("Delete a page of object concurrently")
+			result = util.BatchDeleteObjectsConcurrently(ctx, s, objs, goroutineConcurrency)
+		}
+
+		if len(result.Deleted) != 0 {
+			klog.Infof("Delete these objects for cluster successfully: %s", strings.Join(result.Deleted, ","))
```

Comment:
too many logs?

Reply:
I think log is less than before, or remove logs about deleted objects ?